obspy/signal/array_analysis.py

```diff
@@ -763,18 +762,15 @@ def array_transff_wavenumber(coords, klim, kstep, coordsys='lonlat'):
     nkx = int(np.ceil((kxmax + kstep / 10. - kxmin) / kstep))
     nky = int(np.ceil((kymax + kstep / 10. - kymin) / kstep))
 
-    transff = np.empty((nkx, nky))
+    kxgrid,kygrid = np.meshgrid(np.linspace(kxmin, kxmax, nkx),
+                                np.linspace(kymin, kymax, nky))
 
-    for i, kx in enumerate(np.arange(kxmin, kxmax + kstep / 10., kstep)):
-        for j, ky in enumerate(np.arange(kymin, kymax + kstep / 10., kstep)):
-            _sum = 0j
-            for k in range(len(coords)):
-                _sum += np.exp(complex(0.,
-                               coords[k, 0] * kx + coords[k, 1] * ky))
-            transff[i, j] = abs(_sum) ** 2
+    ks  = np.transpose(np.vstack( (kxgrid.flatten(),kygrid.flatten()) ))
 
-    transff /= transff.max()
-    return transff
+    k_dot_r = np.einsum('ni,mi->nm',ks,coords[:,:2]) #z coordinate is not used
+    transff = np.abs(np.sum(np.exp(1j*k_dot_r), axis=1))**2/len(coords)**2
```

Comment:
Please use PEP8 style; a space is required around operators and after commas and comment characters.